added ob-raku recipe

masukomi opened the pull request:
### Brief summary of what the package does

provides org-babel with support for the [Raku](https://raku.org/)  language

### Direct link to the package repository

https://github.com/masukomi/ob-raku

### Your association with the package

Maintainer. Creator has deleted original repo, code was lost to mailing list archives. I've revived it, given it a test, and provided a public repo for it using the original license.

As best I can tell, my repo is the only version of this in any public code repository.

### Relevant communications with the upstream package maintainer

**None needed** 


### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
  - NOTE: specifically did not address linter complaint about `org-babel-expand-body:raku` as that seems to be the
    convention in org-babel and I'm not sure if changing it would break things.
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.

I also found https://github.com/ohmycloud/ob-raku after a very quick search; since you mentioned not being any other version in a public repository, is there anything in the ohmycloud repo that might've been missed?

Other quick notes:

- Yes, the org-babel package-name/function-name idiosyncrasy is fine.
- For your recipe, as long as `main` is your default branch, you should avoid specifying `:branch` (i.e. prefer the default recipe).
- ob-raku.el#L166: It's safer to sharp-quote function names; use `#'`

masukomi commented:
> ob-raku.el#L166: It's safer to sharp-quote function names; use #'

changed. 


I diffed the version at https://github.com/ohmycloud/ob-raku

it appears to be exactly what the comments implied. A slightly older version of the same thing. Tim's 0.05 version, which mine is based on (same thing + renames as per linter & melpa suggestions) appears slightly more refined. Some code refactored out into a separate method but nothing huge. 

the additional stuff in my repo

* 1.0.1 --- minor correction to line that became uncommented
* 1.0.0 --- Applied package-lint suggestions. Switched to Semantic Versioning.
* 0.05 --- Added initial support for parentheses and commas in strings in lists without breaking the lists on return.
* 0.04 --- Added square brackets to list splitting, so as to split embedded arrays as well as lists.

riscy commented:
Noted; I'll just tag @ohmycloud as the repo owner and @tmtvl as the only contributor to that repo so they're aware we're talking about them.  Merging!
